Tests/Engine/samplerule/samplerule.psm1

```diff
@@ -28,9 +28,12 @@ function Measure-RequiresRunAsAdministrator
         [System.Management.Automation.Language.ScriptBlockAst]
         $testAst
     )        
-        $dr = New-Object `
+    $l=(new-object System.Collections.ObjectModel.Collection["Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent"])
```

Comment:
in PowerShellCore tests, we're moving away from using new-object in favor of this:
```
$l = [System.Collections.ObjectModel.Collection[Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent]]::new()
$c = [Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent]::new(1,2,3,4,'text','filePath','description')
$null = $l.Add($c)
$dr = [Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.DiagnosticRecord]::new(
    "This is help", 
    $ast.Extent,$PSCmdlet.MyInvocation.InvocationName,
    "Warning",
    $null,
    $null,
    $l)
return $dr
```
which I think is much tidier. I know it's inconsistent, but that's ok - we can go back where we need to.

Reply:
OK, I changed it to that and updated the docs as well. I guess the performance of that is also better because it does not have to go through the layer of a cmdlet.

Reply:
yah - I thought about this again last night and realized this wouldn't work because of the downlevel requirements. We won't be able to support the modern constructor syntax until we drop support for ps v3/4